File: VB-Examples/07_DocumentOperation/GetVariables.vb
Imports System.ComponentModel
Imports System.IO
Imports System.Text
Imports Spire.Doc
Imports Spire.Doc.Documents

Namespace GetVariables
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new instance of the Document class
			Dim document As New Document()

			' Load the specified document file
			document.LoadFromFile("..\..\..\..\..\..\..\Data\Template_Docx_6.docx")

			' Initialize a StringBuilder to store the content
			Dim stringBuilder As New StringBuilder()

			' Add initial message to the StringBuilder
			stringBuilder.AppendLine("This document has following variables:")

			' Iterate through each variable in the document and append their name and value to the StringBuilder
			For Each entry As KeyValuePair(Of String, String) In document.Variables
				Dim name As String = entry.Key
				Dim value As String = entry.Value
				stringBuilder.AppendLine("Name: " & name & ", " & "Value: " & value)
			Next entry

			' Write the content of the StringBuilder to a text file
			File.WriteAllText("GetVariables_out.txt", stringBuilder.ToString())

			' Dispose of the document object
			document.Dispose()

			WordDocViewer("GetVariables_out.txt")
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub


	End Class
End Namespace

File: CS-Examples/07_DocumentOperation/GetVariables.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Text;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;

namespace GetVariables
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new Document object
			Document document = new Document();

			// Load a Word document from the specified file path
			document.LoadFromFile(@"..\..\..\..\..\..\..\Data\Template_Docx_6.docx");

			// Create a StringBuilder to hold the content for the output file
			StringBuilder stringBuilder = new StringBuilder();

			// Append an introductory line to the StringBuilder
			stringBuilder.AppendLine("This document has following variables:");

			// Iterate through each key-value pair in the document's Variables collection
			foreach (KeyValuePair<string, string> entry in document.Variables)
			{
				// Extract the name and value from the current entry
				string name = entry.Key;
				string value = entry.Value;

			// Append the name and value to the StringBuilder
			stringBuilder.AppendLine("Name: " + name + ", " + "Value: " + value);
			}

			// Specify the file name for the saved output file
			string result = "GetVariables_out.txt";

			// Write the content of the StringBuilder to a text file
			File.WriteAllText(result, stringBuilder.ToString());

			// Release the resources used by the document
			document.Dispose();

            WordDocViewer("GetVariables_out.txt");
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        
    }
}
